lib/Sema/BuilderTransform.cpp

```diff
@@ -169,8 +169,15 @@ class BuilderClosureVisitor
       }
 
       auto expr = node.get<Expr *>();
-      if (wantExpr)
-        expr = new (ctx) OneWayExpr(expr);
+      if (wantExpr) {
+        if (builderSupports(ctx.Id_buildExpression)) {
+          expr = buildCallIfWanted(expr->getLoc(), ctx.Id_buildExpression,
+                                   { expr }, { Identifier() },
+                                   /*allowOneWay=*/true);
```

Comment:
I know the parameter name is existing code, but why is this "allow" one-way, since it's an unconditional instruction to make the result one-way?

Anyway, this seems like the right interaction with one-way, although arguably it would be clearer in code if that was done unconditionally as a separate step.

Reply:
It used to be conditional, and I just removed the staging flag. I can clean this up